Source/ActivityIndicator/ActivityIndicator.swift

```diff
@@ -50,20 +49,29 @@ public final class ActivityIndicator<T>: ObservableConvertibleType {
             let messageChangeDisposable = observable.map(associatedValueProvider)
                 .startWith(initialAssociatedValue)
                 .distinctUntilChanged(self.equalFunction)
-                .subscribe(onNext: {
-                    if let index = self.variable.value.index(where: { $0.id == id }) {
-                        self.variable.value[index].associatedValue = $0
+                .subscribe(onNext: { associatedValue in
+                    if let index = self.behaviorRelay.value.index(where: { $0.id == id }) {
+                        self.behaviorRelay.mutate(using: {
+                            $0[index].associatedValue = associatedValue
+                        })
                     } else {
-                        self.variable.value.append((id: id, associatedValue: $0))
+                        self.behaviorRelay.mutate(using: {
+                            $0.append((id: id, associatedValue: associatedValue))
+                        })
                     }
                 })
             
             return Disposables.create {
                 subscriptionDisposable.dispose()
                 messageChangeDisposable.dispose()
                 
-                if let index = self.variable.value.index(where: { $0.id == id }) {
-                    self.variable.value.remove(at: index)
+                if let index = self.behaviorRelay.value.index(where: { $0.id == id }) {
+                    self.behaviorRelay.mutate(using: {
+                        $0.remove(at: index)
+                    })
+                    var mutableArray = self.behaviorRelay.value
```

Comment:
What is the difference between the `mutate` and removing it using `mutableArray`?

Reply:
I overlooked it, it should have been deleted.